src/store/reducers/signinSlice.js

```diff
@@ -0,0 +1,49 @@
+import  {createSlice} from "@reduxjs/toolkit";
+
+const signinSlice = createSlice({
+    name: 'loging',
+
+    initialState: {        
+        submitted: sessionStorage.getItem('submitted'),
+        username: sessionStorage.getItem('username'),
+        password: '',
+        userrole: sessionStorage.getItem('userrole')
+    },
+
+    reducers: {
+        signin: (state, action) => {
+            const user = action.payload;            
+            state.submitted = user.signedin;
+            
+            state.username = user.username;
+            state.password = user.password;
+
+            if (state.username === 'testAdmin@gmail.com' && state.password === '12345yuiopp') {
+                state.userrole = 'Admin';
+            } else {
+                state.userrole = 'User';
+            }
+
+            sessionStorage.setItem('submitted', user.signedin);
```

Comment:
Why you ever need 'submitted' flag? 'username' exists ==> submitted

Reply:
done